Code/Tools/FBuild/FBuildCore/Graph/NodeGraph.cpp

```diff
@@ -1436,6 +1436,77 @@ Node * NodeGraph::FindNodeInternal( const AString & fullPath ) const
 	return nullptr;
 }
 
+// FindNearestNodesInternal
+//------------------------------------------------------------------------------
+size_t NodeGraph::FindNearestNodesInternal( const AString & fullPath, const Node **nodes, uint32_t capacity ) const
+{
+	ASSERT( Thread::IsMainThread() );
+	ASSERT( nodes );
+	ASSERT( capacity );
+
+	if ( fullPath.IsEmpty() )
+		return 0;
+
+	const uint32_t maxDistance = 5;
+	uint32_t worstMinDistance = fullPath.GetLength() + 1;
+
+	ASSERT( capacity <= 8 );
```

Comment:
You could also use an Array<> here, presized to the capacity of the input vector.